### DISCUSSION TITLE
Docs of functions: is this really a closure?

### DISCUSSION THREAD

**Discussion by waleed-alfaifi** [mdn/mdn-community#588] [Content] [answered]
Hey everyone,

I was reading the docs of functions in JavaScript and I ran into this example:
```js
function addSquares(a, b) {
  function square(x) {
    return x * x;
  }
  return square(a) + square(b);
}
a = addSquares(2, 3); // returns 13
b = addSquares(3, 4); // returns 25
c = addSquares(4, 5); // returns 41
```

The docs say that the inner function `square` forms a closure with its enclosing function `addSquares`.
> Since the inner function forms a closure, you can call the outer function and specify arguments for both the outer and inner function.

As far as I know, closures are created only if the nested function is referencing variables of the outer function. However, in the above example, `square` is not referencing any variables/functions of `addSquares` and it's just sitting there in the context of `addSquares`.

I was going to open a PR for this, but I wanted to make sure I understand it well.

**Comment by ghost** [comment, marked as answer]
You should be correct - neither of these functions enclose anything from an outer scope.
It is correct in that you *could* call the outer function from it, but as it does not reference the function within the body, it does not close over it.

**Comment by shashiranjan999** [comment]
you are correct according to the browser devtool till now we are not accessing any variable from outer scope it will not form closure

attach image from reference if i am wrong please correct me.you can see in image inner function not create closure with outer function

<img width="360" alt="image" src="https://github.com/mdn/mdn-community/assets/93181489/68086ab9-b633-4a1a-b074-14e53d2d5bcf">

**Reply by Shubham962546** [reply]
Great finding bro have faced same issue . should be rectified
